src/OrchardCore/OrchardCore.ContentManagement.GraphQL/Queries/Types/DynamicContentTypeBuilder.cs

```diff
@@ -84,25 +92,11 @@ public void Build(FieldType contentQuery, ContentTypeDefinition contentTypeDefin
                 else
                 {
                     // Check if another builder has already added a field for this part.
-                    var existingField = contentItemType.GetField(partName.ToFieldName());
-                    if (existingField != null)
-                    {
-                        // Add content field types.
-                        foreach (var field in part.PartDefinition.Fields)
-                        {
-                            foreach (var fieldProvider in contentFieldProviders)
-                            {
-                                var contentFieldType = fieldProvider.GetField(field, part.Name);
-
-                                if (contentFieldType != null && !contentItemType.HasField(contentFieldType.Name))
-                                {
-                                    contentItemType.AddField(contentFieldType);
-                                    break;
-                                }
-                            }
-                        }
+                    // We do not modify the existing field here again. The other builder
+                    // must take care to fully populate the part type.
+                    var partFieldName = partName.ToFieldName();
```

Comment:
If we want to remove this logic we need to make sure that the problem of #5536 doesn't come back.

Reply:
Adding a unit test is something useful

Reply:
You are absolutely right, I cannot remove this.

Reply:
I think I tackled this in the correct way now.